dGame/dComponents/TriggerComponent.cpp

```diff
@@ -113,22 +160,169 @@ std::vector<Entity*> TriggerComponent::GatherTargets(LUTriggers::Command* comman
 
 	if (command->target == "self") entities.push_back(m_Parent);
 	else if (command->target == "zone") { /*TODO*/ }
-	else if (command->target == "target") { /*TODO*/ }
-	else if (command->target == "targetTeam") { /*TODO*/ }
-	else if (command->target == "objGroup") entities = EntityManager::Instance()->GetEntitiesInGroup(command->targetName);
-	else if (command->target == "allPlayers") { /*TODO*/ }
-	else if (command->target == "allNPCs") { /*TODO*/ }
-
-	if (optionalTarget) entities.push_back(optionalTarget);
+	else if (command->target == "target" && optionalTarget) entities.push_back(optionalTarget);
+	else if (command->target == "targetTeam" && optionalTarget) {
+		auto* team = TeamManager::Instance()->GetTeam(optionalTarget->GetObjectID());
+		for (const auto memberId : team->members) {
+			auto* member = EntityManager::Instance()->GetEntity(memberId);
+			if (member) entities.push_back(member);
+		}
+	} else if (command->target == "objGroup") entities = EntityManager::Instance()->GetEntitiesInGroup(command->targetName);
+	else if (command->target == "allPlayers") {
+		for (auto* player : Player::GetAllPlayers()) {
+			entities.push_back(player);
+		}
+	} else if (command->target == "allNPCs") { /*UNUSED*/ }
 
 	return entities;
 }
 
-void TriggerComponent::HandleSetPhysicsVolume(Entity* targetEntity, std::vector<std::string> argArray, std::string target) {
-	PhantomPhysicsComponent* phanPhys = m_Parent->GetComponent<PhantomPhysicsComponent>();
-	if (!phanPhys) return;
+void TriggerComponent::HandleFireEvent(Entity* targetEntity, std::string args) {
+	for (CppScripts::Script* script : CppScripts::GetEntityScripts(targetEntity)) {
+		script->OnFireEventServerSide(targetEntity, m_Parent, args, 0, 0, 0);
+	}
+}
+
+void TriggerComponent::HandleDestroyObject(Entity* targetEntity, std::string args){
+	uint32_t killType;
+	GeneralUtils::TryParse<uint32_t>(args, killType);
+	targetEntity->Smash(m_Parent->GetObjectID(), static_cast<eKillType>(killType));
+}
+
+void TriggerComponent::HandleToggleTrigger(Entity* targetEntity, std::string args){
+	auto* triggerComponent = targetEntity->GetComponent<TriggerComponent>();
+	if (!triggerComponent) return;
+	triggerComponent->SetTriggerEnabled(args == "1");
+}
+
+void TriggerComponent::HandleResetRebuild(Entity* targetEntity, std::string args){
+	auto* rebuildComponent = targetEntity->GetComponent<RebuildComponent>();
+	if (!rebuildComponent) return;
+	rebuildComponent->ResetRebuild(args == "1");
+}
+
+void TriggerComponent::HandleMoveObject(Entity* targetEntity, std::vector<std::string> argArray){
+	auto position = targetEntity->GetPosition();
+
+	NiPoint3 offset = NiPoint3::ZERO;
+	GeneralUtils::TryParse<float>(argArray.at(0), offset.x);
+	GeneralUtils::TryParse<float>(argArray.at(1), offset.y);
+	GeneralUtils::TryParse<float>(argArray.at(2), offset.z);
+
+	position.x += offset.x;
+	position.y += offset.y;
+	position.z += offset.z;
```

Comment:
`position + offset`
NiPoints have their operator+= overloaded iirc but if they don't it would make more sense to implement that than to have to type these three lines every time